sdk/highlight-java/highlight-sdk/src/main/java/io/highlight/sdk/HighlightOpenTelemetry.java

```diff
@@ -0,0 +1,109 @@
+package io.highlight.sdk;
+
+import java.util.Arrays;
+
+import io.highlight.sdk.common.HighlightAttributes;
+import io.highlight.sdk.common.HighlightOptions;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.logs.Logger;
+import io.opentelemetry.api.trace.TracerProvider;
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.exporter.otlp.http.logs.OtlpHttpLogRecordExporter;
+import io.opentelemetry.exporter.otlp.http.trace.OtlpHttpSpanExporter;
+import io.opentelemetry.extension.trace.propagation.B3Propagator;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.logs.LogRecordProcessor;
+import io.opentelemetry.sdk.logs.SdkLoggerProvider;
+import io.opentelemetry.sdk.logs.export.BatchLogRecordProcessor;
+import io.opentelemetry.sdk.logs.export.LogRecordExporter;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+
+public class HighlightOpenTelemetry implements OpenTelemetry {
+
+	private final SdkTracerProvider tracerProvider;
+	private final SdkLoggerProvider loggerProvider;
+
+	private final ContextPropagators propagator;
+
+	HighlightOpenTelemetry(Highlight highlight) {
+		HighlightOptions options = highlight.getOptions();
+
+		AttributesBuilder attributesBuilder = Attributes.builder()
+				.put(HighlightAttributes.HIGHLIGHT_PROJECT_ID, options.projectId())
+				.put(ResourceAttributes.TELEMETRY_SDK_LANGUAGE, "java") // TODO compile with maven
+				.put(ResourceAttributes.TELEMETRY_SDK_NAME, "highlight") // TODO compile with maven
```

Comment:
what does `compile with maven` mean? do you mean the `highlight` constant can come from maven?

Reply:
Maven can automaticly insert the currently build version from there pom.xml